datafusion/src/logical_plan/plan.rs

```diff
@@ -50,6 +52,28 @@ pub enum JoinType {
     Anti,
 }
 
+impl FromStr for JoinType {
+    type Err = DataFusionError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "inner" => Ok(JoinType::Inner),
+            "left" => Ok(JoinType::Left),
+            "right" => Ok(JoinType::Right),
+            "full" => Ok(JoinType::Full),
+            "semi" => Ok(JoinType::Semi),
+            "anti" => Ok(JoinType::Anti),
+            how => {
+                return Err(DataFusionError::Internal(format!(
```

Comment:
This is not an internal error, as people would be able to write anything from Python.